### ISSUE TITLE
Unknown fields sorted as literal

### ISSUE THREAD

**Issue by moewew** [plk/biber#262]
Inspired by the discussion in https://github.com/plk/biblatex/issues/884.

When run as-is, the example
```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage{filecontents}
\begin{filecontents}{foo.dbx}
\ProvidesFile{foo.dbx}[2019/04/27 v0.1 unknown field sort]
\DeclareDatamodelFields[type=field, datatype=literal]{
  foo}
\end{filecontents}

\usepackage[style=authoryear, backend=biber, datamodel=foo]{biblatex}

\DeclareSortingTemplate{foo}{
  \sort{
    \field{presort}
  }
  \sort[final]{
    \field{sortkey}
  }
  \sort{
    \field{note}
    \field{foo}
  }
}

\ExecuteBibliographyOptions{sorting=foo}


\DeclareFieldFormat{note}{\texttt{note}: #1}
\DeclareFieldFormat{foo}{\texttt{foo}: #1}
\DeclareFieldFormat{title}{\texttt{title}: #1}
% just to make things clearer
\DeclareBibliographyDriver{misc}{%
  \usebibmacro{bibindex}%
  \usebibmacro{begentry}%
  \printfield{note}%
  \newunit\newblock
  \printfield{foo}%
  \newunit\newblock
  \printfield{title}%
  \usebibmacro{finentry}}

\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@misc{a,
  note    = {A},
}
@misc{d,
  note    = {D},
}
@misc{e,
  note    = {E},
}
@misc{g,
  note    = {G},
}
@misc{none,
  title   = {Can't have this entry totally empty},
}
\end{filecontents}

\addbibresource{\jobname.bib}

\begin{document}
\nocite{*}
\printbibliography
\end{document}
```
produces

>title: Can’t have this entry totally empty.
>
> note: A.
>
> note: D.
>
> note: E.
>
> note: G.

But when one removes the `datamodel=foo` from the `biblatex` option (and turns `foo` into an unknown field) the sorting becomes

> note: A.
>
> note: D.
>
> note: E.
>
> title: Can’t have this entry totally empty.
>
> note: G.

It appears that the entry without `note` is sorted under `F` for `foo`.

I suspect the `\field{foo}` is more or less treated as `\literal{foo}` for unknown fields. There is no warning on the Biber run about this.

1. Would it be possible to warn about an unknown field passed to `\field` on a Biber run?
2. Would it be possible to just ignore unknown fields and not treat them as `\literal` for sorting?

**Comment by plk** [comment]
Done in 2.13 - there is a warning and such fields not in the data model are just skipped. This requires an update to biblatex, done in 3.13 DEV.

**Comment by moewew** [comment]
Works very well. Thank you.

**Comment by moewew** [comment]
Seems as though the order of the conditionals in `Internals.pm` means that `\literal{<string>}` is treated as `\field{<string>}` if `<string>` is a valid field name.
```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage{filecontents}
\begin{filecontents}{foo.dbx}
\ProvidesFile{foo.dbx}[2019/04/27 v0.1 unknown field sort]
\DeclareDatamodelFields[type=field, datatype=literal]{
  foo}
\end{filecontents}

\usepackage[style=authoryear, backend=biber, datamodel=foo]{biblatex}

\DeclareSortingTemplate{foo}{
  \sort{
    \field{presort}
  }
  \sort[final]{
    \field{sortkey}
  }
  \sort{
    \field{note}
    \literal{foo}
  }
}

\ExecuteBibliographyOptions{sorting=foo}


\DeclareFieldFormat{note}{\texttt{note}: #1}
\DeclareFieldFormat{foo}{\texttt{foo}: #1}
\DeclareFieldFormat{title}{\texttt{title}: #1}
% just to make things clearer
\DeclareBibliographyDriver{misc}{%
  \usebibmacro{bibindex}%
  \usebibmacro{begentry}%
  \printfield{note}%
  \newunit\newblock
  \printfield{foo}%
  \newunit\newblock
  \printfield{title}%
  \usebibmacro{finentry}}

\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@misc{a,
  note    = {A},
}
@misc{d,
  note    = {D},
}
@misc{e,
  note    = {E},
}
@misc{g,
  note    = {G},
}
@misc{none,
  title   = {Can't have this entry totally empty},
}
\end{filecontents}

\addbibresource{\jobname.bib}

\begin{document}
\nocite{*}
\printbibliography
\end{document}
```
this time with **`\literal{foo}`** instead of `\field{foo}` shows the effect. It produces

>title: Can’t have this entry totally empty.
>
> note: A.
>
> note: D.
>
> note: E.
>
> note: G.

But I would like to see

> note: A.
>
> note: D.
>
> note: E.
>
> title: Can’t have this entry totally empty.
>
> note: G.

**Comment by plk** [comment]
Yes, this was a rather silly oversight. Fixed in DEV.

**Comment by moewew** [comment]
Thanks, that fixed it.

**Comment by plk** [comment]
Fixed in latest 2.13 release.
